src/gui/shellextensionsserver.cpp

```diff
@@ -16,29 +16,58 @@
 #include "account.h"
 #include "accountstate.h"
 #include "common/shellextensionutils.h"
+#include "libsync/vfs/cfapi/shellext/configvfscfapishellext.h"
 #include "folder.h"
 #include "folderman.h"
+#include "ocssharejob.h"
 #include <QDir>
-#include <QJsonDocument>
+#include <QJsonArray>
+#include <QJsonObject>
 #include <QLocalSocket>
 
+namespace {
+constexpr auto isSharedInvalidationInterval = 2 * 60 * 1000; // 2 minutes, so we don't make fetch sharees requests too often
+constexpr auto folderAliasPropertyKey = "folderAlias";
+}
+
 namespace OCC {
 
 ShellExtensionsServer::ShellExtensionsServer(QObject *parent)
     : QObject(parent)
 {
+    _isSharedInvalidationInterval = isSharedInvalidationInterval;
     _localServer.listen(VfsShellExtensions::serverNameForApplicationNameDefault());
     connect(&_localServer, &QLocalServer::newConnection, this, &ShellExtensionsServer::slotNewConnection);
 }
 
 ShellExtensionsServer::~ShellExtensionsServer()
 {
+    {
+        QMutexLocker locker(&_customStateSocketConnectionsMutex);
```

Comment:
why do you need mutex lockers ?

Reply:
@mgallien `_customStateSocketConnections` is being modified from 2 threads (local socket connection thread and network job to fetch shares)